### PR TITLE
Allow sending Altair blocks to `/eth/v1/beacon/blocks`

### LINKED ISSUE #9669: POST /eth/v1/beacon/blocks with Altair block gives error
# 🐞 Bug Report

### Description

POSTing a block to the `/eth/v1/beacon/blocks` endpoint returns an error.

### Has this worked before in a previous version?

No.

## 🔬 Minimal Reproduction

POST a block to the relevant endpoint on prater.

## 🔥 Error

<pre><code>
{"message":"Could not process beacon block: could not process block: could not execute state transition: could not process block: state and block are different version. 1 != 0","code":500}
</code></pre>

## 🌍  Your Environment

develop.

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

A bit of explanation on what happened in `custom_hooks.go`.

`setInitialPublishBlockPostRequest` is triggered before we deserialize the request JSON into a struct. We don't know which version of the block got posted, but we can determine it from the slot. We know that both Phase 0 and Altair blocks have a `Message` field with a `Slot` field, so we deserialize the request into a struct `s`, which has the right fields, to obtain the slot. Once we know the slot, we can determine what the `PostRequest` field of the endpoint should be, and we set it appropriately.

In `preparePublishedBlock` we transform the `PostRequest`. gRPC expects either a `phase0_block` or a `altair_block` field in the JSON object, but we have a `message` field at this point. We do a simple conversion depending on the type of `endpoint.PostRequest` (which was filled out previously in `setInitialPublishBlockPostRequest`)

**Which issues(s) does this PR fix?**

Fixes #9669

**Other**

This PR also makes sure that `AltairE2EForkEpoch` constant is not publicly visible any more. Its misuse caused a bug in the initial implementation.

### CODE REVIEW

**Review by rauljordan — DISMISSED** [review]

File: beacon-chain/rpc/apimiddleware/custom_hooks.go [hunk 1]
```diff
@@ -168,15 +170,82 @@ func wrapSignedContributionAndProofsArray(
...
+		return false, apimiddleware.InternalServerErrorWithMessage(err, "could not read body")
+	}
+	if err := json.Unmarshal(buf, &s); err != nil {
+		return false, apimiddleware.InternalServerErrorWithMessage(err, "could not read slot from body")
+	}
+	slot, err := strconv.ParseUint(s.Message.Slot, 10, 64)
+	if err != nil {
+		return false, apimiddleware.InternalServerErrorWithMessage(err, "slot is not an unsigned integer")
+	}
+	if slot < uint64(params.AltairE2EForkEpoch) {
```
Comment (rauljordan) [inline beacon-chain/rpc/apimiddleware/custom_hooks.go:209]: e2e fork epoch is only used in e2e

File: beacon-chain/rpc/apimiddleware/custom_hooks.go [hunk 2]
```diff
@@ -168,15 +170,82 @@ func wrapSignedContributionAndProofsArray(
...
+		return false, apimiddleware.InternalServerErrorWithMessage(err, "could not read body")
+	}
+	if err := json.Unmarshal(buf, &s); err != nil {
+		return false, apimiddleware.InternalServerErrorWithMessage(err, "could not read slot from body")
+	}
+	slot, err := strconv.ParseUint(s.Message.Slot, 10, 64)
+	if err != nil {
+		return false, apimiddleware.InternalServerErrorWithMessage(err, "slot is not an unsigned integer")
+	}
+	if slot < uint64(params.BeaconConfig().AltairForkEpoch) {
```
Comment (terencechain) [inline beacon-chain/rpc/apimiddleware/custom_hooks.go:209]: We are comparing slot against epoch. Should we cover this with a unit test?

**Review by terencechain — CHANGES_REQUESTED** [review]

File: beacon-chain/rpc/apimiddleware/custom_hooks.go (hunk 2 above)
Comment (rkapka) [reply beacon-chain/rpc/apimiddleware/custom_hooks.go]: I added missing tests for `setInitialPublishBlockPostRequest`.

File: testing/endtoend/helpers/helpers.go [hunk 3]
```diff
@@ -33,6 +33,7 @@ const (
 	cpuProfileFileName  = "node_cpu_profile_%d.pb.gz"
 	fileBufferSize      = 64 * 1024
 	maxFileBufferSize   = 1024 * 1024
+	AltairE2EForkEpoch  = 6
```
Comment (terencechain) [inline testing/endtoend/helpers/helpers.go:36]: Seems out of the scope for this PR but can't hurt, Please update the PR description though!

**Review by terencechain — APPROVED** [review]

**Pull request merged**